crates/sncast/src/starknet_commands/account/import.rs

```diff
@@ -58,36 +57,33 @@ pub struct Add {
     pub rpc: RpcArgs,
 }
 
-pub async fn add(
+pub async fn import(
     account: &str,
     accounts_file: &Utf8PathBuf,
     provider: &JsonRpcClient<HttpTransport>,
-    add: &Add,
-) -> Result<AccountAddResponse> {
-    let private_key = match &add.private_key_file_path {
-        Some(file_path) => get_private_key_from_file(file_path)
-            .with_context(|| format!("Failed to obtain private key from the file {file_path}"))?,
-        None => add
-            .private_key
-            .expect("Failed to parse provided private key"),
+    import: &Import,
+) -> Result<AccountImportResponse> {
+    let private_key = match (&import.private_key, &import.private_key_file_path) {
+        (Some(passed_private_key), None) => passed_private_key,
+        (None, Some(passed_private_key_file_path)) => &{
+            get_private_key_from_file(passed_private_key_file_path).with_context(|| {
+                format!("Failed to obtain private key from the file {passed_private_key_file_path}")
+            })?
+        },
+        (None, None) => &get_private_key_from_input(),
+        _ => unreachable!("Checked on clap level"),
     };
```

Comment:
I'm not a fan of this construct forcing matches like this, why not write it as `if let(...) = {} else if let (...) = {} else {}`

Reply:
Fine, refactored.